### PR TITLE
Va 1618 updating vod model

### PR DESCRIPTION
**Pull request by howardrigberg**
#### Ticket
[VA-1618](https://vimean.atlassian.net/browse/VA-1618)

#### Ticket Summary
Updating model for VOD support.

#### Implementation Summary
Updating project gradle version. Adding filters for subscriptions, rentals, and purchases. Adding equals and hashCode support to VodItem object.

#### How to Test
Ensure that networking continues to function as expected.

### CODE REVIEW

File: vimeo-networking/src/main/java/com/vimeo/networking/model/VodItem.java [hunk 1]
```diff
@@ -206,4 +201,22 @@ public void setFilm(@Nullable Video film) {
...
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || !(obj instanceof VodItem)) {
+            return false;
+        }
+        VideoFile that = (VideoFile) obj;
+
+        return (mLink != null && that.getLink() != null) && mLink.equals(that.getLink());
```
Comment (kevinzetterstrom) [inline vimeo-networking/src/main/java/com/vimeo/networking/model/VodItem.java]: Comparing a VodItem vs a VideoFile???

File: vimeo-networking/src/main/java/com/vimeo/networking/model/VodItem.java [hunk 2]
```diff
@@ -128,11 +128,6 @@ public String getDescription() {
     }
 
     @Nullable
-    public String getLink() {
-        return mLink;
-    }
```
Comment (kevinzetterstrom) [inline vimeo-networking/src/main/java/com/vimeo/networking/model/VodItem.java]: Why remove this getter?

**Comment by howardrigberg** [comment]
Worst. Code. Ever. Nice catch. It was some dumb copy pasta stuff. Here's the mod:

https://github.com/vimeo/vimeo-networking-java/commit/2b3cd0585669f0bfaf06785f3f8f27920a1393ea

**Comment by kevinzetterstrom** [comment]
👍  merge when ready

**Pull request merged**